docs/testing.md

```diff
@@ -0,0 +1,130 @@
+## Purpose
+
+This document describes our approach to testing for the 10101 project.
+
+## Assumptions
+
+Here we list a non-exhaustive set of assumptions that influenced our findings:
+
+### Common Rust code between `app` and `coordinator`
+
+There will be a number of crates that will be used as dependencies in both `app` and `coordinator`.
+
+### No initial effort to keep different versions of `app` and `coordinator` compatible
+
+The `coordinator` will not be backwards-compatible with older versions of the `app` until later in the development of 10101.
+This means that we will not devote effort to test backwards compatibility between these two systems.
+
+### The `coordinator` is treated as a single, multi-purpose system
+
+In the past, we have modularised `coordinator`-like systems into several components for architectural and strategic reasons.
+That may very well still be the case for 10101, but at this stage we lack the insight to split this system.
+Once that design has been completed we can revise this document accordingly.
+
+### Common `10101-lightning-node` dependency between `app` and `coordinator`
+
+This `10101-lightning-node` is one of the aforementioned Rust crates shared between these two systems.
+It represents the custom Lightning Node built using (a fork of) `rust-lightning` and `rust-dlc`.
+
+### The `app`'s state will be controlled by its Rust backend
+
+Thanks to this we should be able to write system tests against the `app`'s API without involving the frontend.
+
+### The architecture will be similar to this component diagram
+
+This diagram does include some of the assumptions listed above:
+
+![Assumed component diagram](http://www.plantuml.com/plantuml/proxy?cache=no&src=https://raw.githubusercontent.com/get10101/10101/d7619613324df3ce5c0f0a35d97bf24323f9515b/docs/diagrams/testing-component-diagram.puml)
```

Comment:
🙃 If you use mermaid you can have this code inline :)

Reply:
Unfortunately mermaid does not have component diagrams - I *wanted* to use it but had to fallback to PlantUML :/

Reply:
we might be able to use one of their other diagram types, but so far I did not find a good fit, see:

https://github.com/mermaid-js/mermaid/issues/1462

https://mermaid.js.org/intro/

Reply:
You can create a component diagram using mermaid: https://mermaid.js.org/syntax/c4c.html. 
Eitherway, plantuml is fine.

Reply:
oh sweet, I missed that C4 diagrams include component diagrams! We should switch to mermaid then!